--- cmd/arbol/sysinfo.go
+++ cmd/arbol/sysinfo.go
@@ -9,6 +9,7 @@
	"runtime"
	"strconv"
	"strings"
+	"syscall"
	"time"
 )

@@ -318,17 +319,13 @@

 func getProcesses() string {
	if runtime.GOOS == "linux" {
-		files, err := os.ReadDir("/proc")
+		var info syscall.Sysinfo_t
+		err := syscall.Sysinfo(&info)
		if err == nil {
-			count := 0
-			for _, f := range files {
-				if f.IsDir() {
-					if _, err := strconv.Atoi(f.Name()); err == nil {
-						count++
-					}
-				}
-			}
-			return strconv.Itoa(count)
+			return strconv.Itoa(int(info.Procs))
		}
	}
	out := runCommand("bash", "-c", "ps -ax | wc -l")
